analyzer/queries/queries.go

```diff
@@ -278,10 +278,18 @@ var (
       ON CONFLICT (entity_id, node_id) DO NOTHING`
 
 	ConsensusEntityUpsert = `
-    INSERT INTO chain.entities (id, address) VALUES ($1, $2)
-      ON CONFLICT (id) DO
-      UPDATE SET
-        address = excluded.address`
+    INSERT INTO chain.entities (id, address, start_date) 
+    SELECT
+      $1,
+      $2,
+      (SELECT MIN(time) FROM chain.blocks as blocks
```

Comment:
if a `registerEntity` tx occurs in the same block, we rely on the tx/block `insert` to occur before this one (which is currently the case), but it introduces some code smell. If the order is reversed this column is `null`. 

alternatives considered:
- trigger function: also not ideal since it introduces dependent logic that is hard for devs to find/grok.
- lookup at query time: the join over the txs table is way too slow

Reply:
Good idea, updated to use height instead!